articles/active-directory/develop/msal-net-acquire-token-silently.md

```diff
@@ -21,6 +21,8 @@ ms.custom: "devx-track-csharp, aaddev"
 
 When you acquire an access token using the Microsoft Authentication Library for .NET (MSAL.NET), the token is cached. When the application needs a token, it should first call the `AcquireTokenSilent` method to verify if an acceptable token is in the cache. In many cases, it's possible to acquire another token with more scopes based on a token in the cache. It's also possible to refresh a token when it's getting close to expiration (as the token cache also contains a refresh token).
 
+For the user flows (eg. [acquiring tokens interactively](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-net-acquire-token-silently)), MSAL caches access, refresh, and ID tokens; and `IAccount` are cached. For application flows (eg. [client credentials](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-authentication-flows#client-credentials)), only access tokens are cached (because account and other tokens are user specific).
```

Comment:
Major:

- The first hyperlink is self-referential (links to the doc you're editing).
- The first sentence in the paragraph above this one ("When you acquire an access token...") now needs adjusting - you're adding two other token types and (somewhat confusingly) an interface to the cache contents.
- The `IAccount` mention needs further explanation if it's required to be mentioned here, or the mention should should be dropped. I foresee questions like:
  - Does the cache store an instance of (an implementation of) that interface
  - The IAccount interface is not a token, why is it in the token cache?
  - Should I get the IAccount instance from the token cache the same way I get tokens? If not, what's the right way?

Minor:

- [Don't use e.g.](https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=33667)
- Don't include locale identifiers like `en-us` in hyperlinks (it breaks localized browser redirects).
- Avoid the term "user flows" to help disambiguate from Azure AD B2C user flows. Instead, use something like "authentication flows that require human interaction."

Reply:
Thanks @mmacy for the feedback.

> The first sentence in the paragraph above this one ("When you acquire an access token...") now needs adjusting - you're adding two other token types and (somewhat confusingly) an interface to the cache contents.

This question came from a 1P migration a few times. There are not two token types, but actually clarifying what is stored in the token cache because based on the scenario (authentication flow with a user or client credentials) different things will be stored in the cache. so @jeevanbalanmanoj  is bringing clarity to this issue. 

> The IAccount mention needs further explanation if it's required to be mentioned here, or the mention should should be dropped. I foresee questions like:

we have public API doc on IAccount which I think could be linked here as reference.

Reply:
I will work w/ @jeevanbalanmanoj  to update. thanks.